src/code_6000.c

```diff
@@ -50,16 +48,25 @@ s32 heap_free(s32 size) {
     }
 }
 
-INCLUDE_ASM("code_6000", collision_heap_create);
-
-INCLUDE_ASM("code_6000", collision_heap_malloc);
+s32 collision_heap_create(void) {
+    if ((s32)_heap_create(&D_80268000, 0x18000) == -1) {
+        return -1;
+    }
+    return 0;
+}
 
-#ifdef NON_MATCHING
-s32 collision_heap_free(s32 size) {
-    s32 isBattle = (*gGameStatusPtr)->isBattle;
+s32 collision_heap_malloc(s32 size) {
+    if (!(*gGameStatusPtr)->isBattle) {
+        return _heap_malloc(&D_80268000, size);
+    } else {
+        return _heap_malloc(&D_803DA800, size);
+    }
+}
 
-    return _heap_free((isBattle == 0) ? (&D_80268000) : (&D_803DA800), size);
+s32 collision_heap_free(UNK_PTR data) {
```

Comment:
imo, this shouldn't be UNK_PTR, it should be a generic pointer type (void*), which in C can take in any kind of pointer and thus is used for free, memcpy, etc.